core/src/main/java/org/apache/sling/testing/mock/sling/context/SlingContextImpl.java

```diff
@@ -217,9 +217,9 @@ private void registerInjectActivateServiceByClassName(@NotNull String @NotNull .
     protected void tearDown() {
 
         if (this.request != null) {
-            MockSlingBindings slingBindings = (MockSlingBindings)this.request.getAttribute(SlingBindings.class.getName());
-            if (slingBindings != null) {
-                slingBindings.tearDown();
+            SlingBindings slingBindings = (SlingBindings)this.request.getAttribute(SlingBindings.class.getName());
+            if (slingBindings != null && slingBindings instanceof MockSlingBindings) {
```

Comment:
sonarcube reports: Remove this unnecessary null check; "instanceof" returns false for nulls.